src/NServiceBus.RavenDB/Outbox/RavenDbOutboxStorage.cs

```diff
@@ -0,0 +1,110 @@
+﻿namespace NServiceBus.RavenDB.Outbox
+{
+    using System;
+    using System.Configuration;
+    using System.Threading;
+    using NServiceBus.Features;
+    using NServiceBus.Outbox.RavenDB;
+    using NServiceBus.RavenDB.Internal;
+    using Raven.Client;
+
+    class RavenDbOutboxStorage : Feature
+    {
+        public RavenDbOutboxStorage()
+        {
+            DependsOn<Outbox>();
+            DependsOn<SharedDocumentStore>();
+            RegisterStartupTask<OutboxCleaner>();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            var store =
+                // Trying pulling a shared DocumentStore set by the user or other Feature
+                context.Settings.GetOrDefault<IDocumentStore>(RavenDbSettingsExtensions.DocumentStoreSettingsKey) ?? SharedDocumentStore.Get(context.Settings);
+
+            if (store == null)
+            {
+                throw new Exception("RavenDB is configured as persistence for Outbox and no DocumentStore instance found");
+            }
+
+            ConnectionVerifier.VerifyConnectionToRavenDBServer(store);
+
+            Helpers.SafelyCreateIndex(store, new OutboxRecordsIndex());
+
+            context.Container.ConfigureComponent<Installer>(DependencyLifecycle.InstancePerCall)
+                .ConfigureProperty(c => c.StoreToInstall, store);
+
+            context.Container.ConfigureComponent<OutboxPersister>(DependencyLifecycle.InstancePerCall)
+                .ConfigureProperty(x => x.DocumentStore, store);
+
+            context.Container.ConfigureComponent<OutboxRecordsCleaner>(DependencyLifecycle.InstancePerCall)
+                .ConfigureProperty(x => x.DocumentStore, store);
+        }
+
+        class OutboxCleaner : FeatureStartupTask, IDisposable
+        {
+            Timer cleanupTimer;
+            TimeSpan frequencyToRunDeduplicationDataCleanup;
+            TimeSpan timeToKeepDeduplicationData;
+            public OutboxRecordsCleaner Cleaner { get; set; }
+
+            public void Dispose()
+            {
+                if (cleanupTimer != null)
+                {
+                    cleanupTimer.Dispose();
+                    cleanupTimer = null;
+                }
+            }
+
+            protected override void OnStart()
+            {
+                var configValue = ConfigurationManager.AppSettings.Get("NServiceBus/Outbox/RavenDB/TimeToKeepDeduplicationData");
```

Comment:
the power of the code api is that they can choose to read from appconfig/any other file/registry/database/environment variable. The assumption that "ops want to control this therefore they will want to controlled it via app.config" makes too many assumptions.

The code api meets all the requirements, is easier to test, easier to evolve, easier to deprecate...